### PR TITLE
Tolouse2 agen update

### PR DESCRIPTION
**Pull request by saif-ericsson**
### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: src/functionaltests/resources/InvalidSubscription.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
         "type": "ARTIFACT_1",
         "conditions": [
           {"jmespath": "gav.groupId=='com.mycompany.myproduct'"},
```
Comment (e-pettersson-ericsson) [inline src/functionaltests/resources/InvalidSubscription.json]: Remove the jmespath condition containing gav.

File: src/functionaltests/resources/InvalidSubscription.json [hunk 2]
```diff
@@ -27,6 +28,7 @@
...
```
Comment (e-pettersson-ericsson) [inline src/functionaltests/resources/InvalidSubscription.json]: This also needs to be updated. Remove jmespath expression with gav.

File: src/functionaltests/resources/queryAggregatedObject1.json [hunk 3]
```diff
@@ -7,6 +7,7 @@
         "object.testCaseExecutions.outcome.id":"TC6"
       }
     ],
-    "object.gav.groupId":"com.mycompany.myproduct"
+    "object.gav.groupId":"com.mycompany.myproduct",
```
Comment (e-pettersson-ericsson) [inline src/functionaltests/resources/queryAggregatedObject1.json]: Remove this gav.

File: src/main/java/com/ericsson/ei/services/SubscriptionService.java [hunk 4]
```diff
@@ -113,17 +113,22 @@ public boolean doSubscriptionExist(String subscriptionName) {
     public boolean modifySubscription(Subscription subscription, String subscriptionName) {
         ObjectMapper mapper = new ObjectMapper();
         Document result = null;
+        String query;
         try {
             String stringSubscription = mapper.writeValueAsString(subscription);
             String ldapUserName = (ldapEnabled) ? HttpSessionConfig.getCurrentUser() : "";
-            String query = generateQuery(subscriptionName, ldapUserName);
+            boolean ownerExist = doSubscriptionOwnerExist(subscriptionName);
```
Comment (e-pettersson-ericsson) [inline src/main/java/com/ericsson/ei/services/SubscriptionService.java]: I assume these changes are not part of this PR?

File: src/functionaltests/resources/eiffel_events_for_test.json [hunk 5]
```diff
@@ -52,11 +52,8 @@
         "classifier": ""
       }
       ],
-      "gav": {
-        "version": "1.58.0",
-        "artifactId": "sub-system",
-        "groupId": "com.mycompany.myproduct"
-      }
+
```
Comment (e-pettersson-ericsson) [inline src/functionaltests/resources/eiffel_events_for_test.json]: Shouldn't meta.version be updated from 1.0.0 to 3.0.0? This goes for all Eiffel events we send in our tests.

**Comment by tobiasake** [comment]
Looks good, but you can improve the commit message. I read commit message as "Tolouse version 2 to Agen version". Change it to a more correct commit message, e.g: "Tolouse2Agen version adaptation" or  "Tolouse to Agen version adaptation"
So we have nothing in the main code that need adapted for the new Eiffel protocol version?

File: src/functionaltests/resources/InvalidSubscription.json (hunk 1 above)
Comment (saif-ericsson) [reply src/functionaltests/resources/InvalidSubscription.json]: done

File: src/functionaltests/resources/InvalidSubscription.json (hunk 2 above)
Comment (saif-ericsson) [reply src/functionaltests/resources/InvalidSubscription.json]: done

File: src/functionaltests/resources/eiffel_events_for_test.json (hunk 5 above)
Comment (saif-ericsson) [reply src/functionaltests/resources/eiffel_events_for_test.json]: done

File: src/test/resources/AggregatedOutput.json [hunk 6]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+["{"fileInformation":[{"extension":"jar","classifier":""}],"buildCommand":null,"testCaseExecutions":[{"testCaseFinishEventId":"11109351-41e0-474a-bc1c-f6e81e58a1c9","testCaseStartedTime":1481875925916,"testCaseStartedEventId":"cb9d64b0-a6e9-4419-8b5d-a650c27c59ca","testCaseFinishedTime":1481875935919,"testCase":{"conclusion":"SUCCESSFUL","verdict":"PASSED","tracker":"My Other Test Management System","id":"TC5","uri":"https://other-tm.company.com/testCase/TC5"}}],"id":"6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43","time":1481875891763,"type":"ARTIFACT_1","identity":"pkg:maven/com.mycompany.myproduct/artifact-name@1.0.0"   }"]
```
Comment (e-pettersson-ericsson) [inline src/test/resources/AggregatedOutput.json]: Don't know why but this file looks strange in GitHub GUI. Perhaps some strange formatting applied to it?

File: src/test/resources/MissedNotificationOutput.json [hunk 7]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+["{"subscriptionName": "Subscription_1","AggregatedObject": {"fileInformation": [{"extension": "jar","classifier": ""}],"buildCommand": null,"testCaseExecutions": [{"testCaseFinishEventId": "11109351-41e0-474a-bc1c-f6e81e58a1c9","testCaseStartedTime": 1481875925916,"testCaseStartedEventId": "cb9d64b0-a6e9-4419-8b5d-a650c27c59ca","testCaseFinishedTime": 1481875935919,"testCase": {"conclusion": "SUCCESSFUL","verdict": "PASSED","tracker": "My Other Test Management System","id": "TC5","uri": "https://other-tm.company.com/testCase/TC5"}}],"id": "6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43","time": 1481875891763,"type": "ARTIFACT_1","identity": "pkg:maven/com.mycompany.myproduct/artifact-name@1.0.0"}}"]
```
Comment (e-pettersson-ericsson) [inline src/test/resources/MissedNotificationOutput.json]: Same strange formatting with this file.

Comment (saif-ericsson) [reply src/test/resources/MissedNotificationOutput.json]: Well, I think formatting look like this as it is not valid json and formatted to fulfill the 'missed notification out put'. A story may be created to turn it into valid json and make necessary changes in code.

File: src/functionaltests/resources/eiffel_events_for_test.json (hunk 5 above)
Comment (e-pettersson-ericsson) [reply src/functionaltests/resources/eiffel_events_for_test.json]: This file still needs to be updated in the event version fields.

File: src/main/resources/templates/rulesTemplate.json [hunk 8]
```diff
@@ -7,7 +7,7 @@
         "StartEvent": "YES",
         "IdentifyRules": "[meta.id]",
         "MatchIdRules": {"_id": "%IdentifyRules_objid%"},
-        "ExtractionRules": "{ id : meta.id, type : meta.type, time : meta.time, gav : data.gav, fileInformation : data.fileInformation, buildCommand : data.buildCommand }",
+        "ExtractionRules": "{ id : meta.id, type : meta.type, time : meta.time, purl : data.identity, fileInformation : data.fileInformation, buildCommand : data.buildCommand }",
```
Comment (evasiba-ericsson) [inline src/main/resources/templates/rulesTemplate.json]: in some rules you use purl in some identity. Please change to identity

File: src/test/resources/MissedNotificationOutput.json (hunk 7 above)
Comment (a-breid-ericsson) [reply src/test/resources/MissedNotificationOutput.json]: I think is because the file is compressed and spaces and new lines has been removed. This was strange befor those changes, so better leave it for now.

File: src/test/resources/EiffelSourceChangeCreatedEvent.json [hunk 9]
```diff
@@ -1,7 +1,7 @@
 {
   "meta": {
     "type": "EiffelSourceChangeCreatedEvent",
-    "version": "1.0.0",
+    "version": "4.0.0",
```
Comment (Christoffer-Cortes) [inline src/test/resources/EiffelSourceChangeCreatedEvent.json]: Shouldn't this be 3? =)

File: src/test/resources/test_All_Events.json [hunk 10]
```diff
@@ -420,7 +408,7 @@
   "EiffelSourceChangeCreatedEvent": {
    "meta": {
     "type": "EiffelSourceChangeCreatedEvent",
-    "version": "1.1.0",
+    "version": "4.0.0",
```
Comment (Christoffer-Cortes) [inline src/test/resources/test_All_Events.json]: Same here.

File: src/test/resources/test_events.json [hunk 11]
```diff
@@ -242,17 +234,13 @@
...
-                "groupId": "com.mycompany.myproduct"
-            }
+            "identity":"pkg:maven/com.mycompany.myproduct/artifact-name@1.0.0"
         }
     },
     "event_EiffelSourceChangeCreatedEvent_3": {
         "meta": {
             "type": "EiffelSourceChangeCreatedEvent",
-            "version": "1.0.0",
+            "version": "4.0.0",
```
Comment (Christoffer-Cortes) [inline src/test/resources/test_events.json]: Same here.

File: src/test/resources/test_events_MP.json [hunk 12]
```diff
@@ -242,17 +234,13 @@
...
-        "groupId": "com.mycompany.myproduct"
-      }
+      "identity": "pkg:maven/com.mycompany.myproduct/artifact-name@1.0.0"
     }
   },
   "event_EiffelSourceChangeCreatedEvent_3": {
     "meta": {
       "type": "EiffelSourceChangeCreatedEvent",
-      "version": "1.0.0",
+      "version": "4.0.0",
```
Comment (Christoffer-Cortes) [inline src/test/resources/test_events_MP.json]: Same here.

File: src/test/resources/upStreamInput.json [hunk 13]
```diff
@@ -113,7 +109,7 @@
    {
       "meta":{
          "type":"EiffelSourceChangeCreatedEvent",
-         "version":"1.0.0",
+         "version":"4.0.0",
```
Comment (Christoffer-Cortes) [inline src/test/resources/upStreamInput.json]: Same here.

File: src/test/resources/upStreamInput.json [hunk 14]
```diff
@@ -147,7 +143,7 @@
...
```
Comment (Christoffer-Cortes) [inline src/test/resources/upStreamInput.json]: Same here.

File: src/test/resources/upStreamResultFile.json [hunk 15]
```diff
@@ -113,7 +109,7 @@
                     {
                         "meta": {
                             "type": "EiffelSourceChangeCreatedEvent",
-                            "version": "1.0.0",
+                            "version": "4.0.0",
```
Comment (Christoffer-Cortes) [inline src/test/resources/upStreamResultFile.json]: Same here.

File: src/test/resources/upStreamResultFile.json [hunk 16]
```diff
@@ -146,7 +142,7 @@
                      {
                         "meta": {
                             "type": "EiffelSourceChangeCreatedEvent",
-                            "version": "1.0.0",
+                            "version": "4.0.0",
```
Comment (Christoffer-Cortes) [inline src/test/resources/upStreamResultFile.json]: Same here.

File: src/test/resources/UpstreamEventsForMockedSourceChange.json [hunk 17]
```diff
@@ -2,7 +2,7 @@
     {
     "meta": {
       "type": "EiffelSourceChangeCreatedEvent",
-      "version": "1.0.0",
+      "version": "4.0.0",
```
Comment (Christoffer-Cortes) [inline src/test/resources/UpstreamEventsForMockedSourceChange.json]: Same here.

File: src/test/resources/UpstreamEventsForMockedSourceChange.json [hunk 18]
```diff
@@ -55,7 +55,7 @@
    {
     "meta": {
       "type": "EiffelSourceChangeCreatedEvent",
-      "version": "1.0.0",
+      "version": "4.0.0",
```
Comment (Christoffer-Cortes) [inline src/test/resources/UpstreamEventsForMockedSourceChange.json]: Same here.

File: src/test/resources/UpstreamEventsForSourceChange.json [hunk 19]
```diff
@@ -2,7 +2,7 @@
   {
     "meta": {
       "type": "EiffelSourceChangeCreatedEvent",
-      "version": "1.0.0",
+      "version": "4.0.0",
```
Comment (Christoffer-Cortes) [inline src/test/resources/UpstreamEventsForSourceChange.json]: Same here.

File: src/test/resources/UpstreamEventsForSourceChange.json [hunk 20]
```diff
@@ -46,7 +46,7 @@
...
```
Comment (Christoffer-Cortes) [inline src/test/resources/UpstreamEventsForSourceChange.json]: Same here.

File: src/test/resources/test_All_Events.json [hunk 21]
```diff
@@ -383,7 +371,7 @@
   "EiffelIssueVerifiedEvent": {
    "meta": {
     "type": "EiffelIssueVerifiedEvent",
-    "version": "1.1.0",
+    "version": "4.0.0",
```
Comment (Christoffer-Cortes) [inline src/test/resources/test_All_Events.json]: Same here.

File: src/test/resources/EiffelSourceChangeCreatedEvent.json (hunk 9 above)
Comment (saif-ericsson) [reply src/test/resources/EiffelSourceChangeCreatedEvent.json]: EiffelSourceChangeCreatedEvent is available in 4.0.0
